Documentation/contributors/design-proposals/installed-api.md

```diff
@@ -0,0 +1,282 @@
+# Installed Operator API
+
+Status: Pending
+
+Version: Alpha
+
+Implementation Owner: TBD
+
+## Motivation
+
+Today, information about operators configured to watch a given namespace is surfaced via "copied" `ClusterServiceVersion`s (`CSV`s). This means that for each __<CSV, watched namespace>__ pair, OLM must manage a copied CSV instance. Additionally, `CSV`s often contain large amounts of metadata, most notably a [base64 encoded icon](https://github.com/operator-framework/operator-lifecycle-manager/blob/18b6c0d2edba2534b4726138b225fff066aae99a/manifests/0000_50_olm_03-clusterserviceversion.crd.yaml#L114). As of OpenShift 4.2, there are more than a dozen namespaces that globally installed operators have their `CSV`s copied to, which causes OLM's `CSV` cache (and memory consumption) to grow large (~150MB) with a fairly low number of operators installed (~5). 
+
+Due to these factors, an additional means of providing this information is required.
+
+## Proposal
+
+Define a new read-only API resource, `installed.packages.operators.coreos.com`, that surfaces metadata about an operator installed and configured to watch a namespace. Back that API by an extension api-server that _synthesizes_ those resources on request.
```

Comment:
I’m wary of mixing up our terms here - package is the entity in a catalog that contains many csvs / operators. 

So I would think that, for example, if we keep this as “packages” we would include much more information than the CSV, like updates available in other channels, etc

Reply:
Ah, I now that I’ve read further I see that it does contain more than the CSV :+1: